docs/getting-started/components.mdx

```diff
@@ -0,0 +1,150 @@
+
+Components in Figma might be resembled by one or multiple variants which are variations of a component.
+
+In Flutter, the `TextButton`, `ElevatedButton`, and `OutlineButton` are variants of a _Button_ component.
+In addition, different Flutter states of a button (e.g. `focused` or `hovered`) might resemble in Figma as additional variants.
+
+<Image
+  zoom
+  src="/assets/getting-started/figma-button-component.svg"
+  alt="Figma Material button component with variants"
+  caption="Figma Material button component with variants"
+/>
+
+## Components Vs. Use-cases
+
+Widgetbook features a **_component_** and **_use-case_** approach in which a single component has one or
+multiple use-cases.
+
+A use-case might resemble a variant of a component or just a specific state of the component.
+
+In Widgetbook, developers create a folder tree structure to organize and catalog components.
+
+## Add your first component
+
+Adding components to Widgetbook works differently based on your choice: manual approach or generator approach
+
+<Tabs
+  groupId="approach"
+  values={[
+    { label: 'Generator', value: 'generator' },
+    { label: 'Manual', value: 'manual' },
+  ]}
+>
+  <TabItem value="generator">
+  1- To add component in generator approach, you can start by adding use-cases with the generator is based on the `@WidgetbookUseCase` annotation.
+
+    ```dart
+    import 'package:flutter/material.dart';
+    import 'package:widgetbook_annotation/widgetbook_annotation.dart';
+
+    @WidgetbookUseCase(
+      name: 'with green color',
+      type: Container,
+    )
+    Widget greenContainerUseCase(BuildContext context) {
+      return Column(
+        children: [
+          Container(
+            color: Colors.green,
+          ),
+        ],
+      );
+    }
+    ```
+
+2- The annotation is used on a use-case builder method with a return type of `Widget` and a single required parameter `BuildContext`.
+
+    <Info>
+        You can freely pick the name of the method, but make sure the name of a newly created use-case builder method does not have a name conflict with already existing methods.
+    </Info>
+
+3- In addition, `@WidgetbookUseCase` requires two mandatory parameters `type` and `name` that allow you to customize how the use-case is shown in the navigation of Widgetbook.
+
+    #### `type` property
+    The `type` property is used by the generator to create the component entry within the navigation.
+    As a developer, you want to specify the type of the cataloged component (`Container`) instead of the one that might be returned by the builder method (`Column`).
+
+    #### `name` property
+    The `name` property of the annotation allows you to specify the name of the use-case.
+    In the case of the example, this is `with green color`.
+
+    4- Execute the `build_runner` by running the following command:
+    ```bash
+    flutter pub run build_runner build
+    ```
+    or
+    ```bash
+    flutter pub run build_runner watch
+    ```
+
```

Comment:
Maybe explain in a sentence what the difference to watch is? 

Also: Hot reload is important for our users. We often see the question: How to hot reload with the generator? Hot reload works properly but more context is required (you need to wait for the generator to finish and then trigger the hot-relaod manually). Where shall we put this information? Does a dedicated page make sense?

Reply:
I guess as flutter developer, the difference might be ovious, however, I do agree, we should create a page for that. I have already created a mdx file just to remember to write about this.